### PR TITLE
[SDK-574] Add CertificatePinner

### PR DESCRIPTION
**Pull request by bitPogo**
## Description

As with the interceptors this pulls out the CertificatePinner of the ApiService.

## Motivation and Context

Currently the CertificatePinner is backed into the ApiService and are not testable and setup in a mighty blur of code.
This is part of [SDK-574](https://gesundheitscloud.atlassian.net/browse/SDK-574) and depends on #135.

## How is it being implemented?

As mentioned it pulls out the already existing CertificatePinner, gives them interfaces and refits them to have a proper structure.

## How Has This Been Tested?

Unit tests for the CertificatePinner had been introduced with this PR.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the changelog accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by wmontwe** [comment]
@bitPogo this has now quite some conflicts

File: sdk-core/src/main/java/care/data4life/sdk/network/util/interceptor/BasicAuthorizationInterceptor.kt [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * contact D4L by email to help@data4life.care.
  */
 
-package care.data4life.sdk.network.interceptor
+package care.data4life.sdk.network.util.interceptor
```
Comment (wmontwe) [inline sdk-core/src/main/java/care/data4life/sdk/network/util/interceptor/BasicAuthorizationInterceptor.kt:17]: why have they been moved to util?

Comment (wmontwe) [reply sdk-core/src/main/java/care/data4life/sdk/network/util/interceptor/BasicAuthorizationInterceptor.kt:17]: we could move them later

**Review by wmontwe — APPROVED** [review]
👍

File: sdk-core/src/main/java/care/data4life/sdk/network/util/interceptor/BasicAuthorizationInterceptor.kt (hunk 1 above)
Comment (wmontwe) [reply sdk-core/src/main/java/care/data4life/sdk/network/util/interceptor/BasicAuthorizationInterceptor.kt:17]: I would keep interceptor

Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/network/util/interceptor/BasicAuthorizationInterceptor.kt:17]: Lets make a final decision once we reached #139.

**Pull request merged**